client/src/hooks/graphql/use-with-data.tsx

```diff
@@ -87,12 +87,12 @@ export function useWithData<T>(fetch: () => Promise<T>): UseData<T> {
     setEditedData({ ...data, ...(editedData || {}), ...edits });
   }
 
-  function saveData(update: UpdateFunc<T>) {
+  async function saveData(update: UpdateFunc<T>): Promise<void> {
     if (state.isLoading || state.isSaving || !editedData || !update) {
       return;
     }
     dispatch({ type: LoadingActionType.SAVING, payload: true });
-    update
+    await update
```

Comment:
when you use `await` you don't need `then` (compiler does that for you). So if you're doing the async version of this would be more like

```typescript
const updated = await update.action(editedData);
if(state.isLoading) {
  //...
```

Let's change the name from `update.callback` to `update.action` if what we're doing is passing an arbitrary action that performs some kind of update.